Created product compatibility matrix (#133)
conduktor/conduktor-public-charts: charts/gateway/docs/COMPATIBILITY_MATRIX.md

# Conduktor Gateway Compatibility Matrix
This compatibility matrix is a resource to help you find which versions of Conduktor Gateway work on which version of our Conduktor Gateway Helm Chart.

> > General recommendation is to use the version of Gateway that comes preconfigured with the helm chart. If needed you can adjust the version in your values property according to the supported console version.

Reviewer: ```suggestion
> > In general we recommend you use the version of Gateway that comes preconfigured with the Helm chart. If needed you can adjust the version in your values property according to the supported Gateway version.
```

> > Breaking changes column will only list breaking change in the helmchart! We strongly recommend reading the full [changelog](https://docs.conduktor.io/changelog/) of Conduktor in case there are breaking within the components.

Reviewer: ```suggestion
> > Breaking changes column will only list breaking change in the helmchart! You must review the Conduktor [changelog](https://docs.conduktor.io/changelog/) to determine whether there are breaking changes within the artefacts.
```
